fundamentals/minimal-apis/samples/IResultImplementation/Tests/IResultImplementationTest.cs

```diff
@@ -0,0 +1,77 @@
+using IResultImplementation.Data;
+using IResultImplementation.Models;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace IResultImplementation.Tests
+{
+    [TestClass]
+    public class IResultImplementationTest
+    {
+        public IQueryable<Contact>? Data { get; set; }
+        public Mock<DbSet<Contact>>? MockSet { get; set; }
+        [TestInitialize]
+        public void Setup()
+        {
+            Data = new List<Contact>()
+            {
+
+                new Contact() { Id = 1, Name = "John", Email = "john@tmail.com", PhoneNumber = "1234567890" },
+                new Contact(){ Id = 2, Name = "Jane", Email = "Jane@kam.com", PhoneNumber = "29384736273" },
+                new Contact(){Id = 3, Name = "Kahn", Email = "ema@email.com", PhoneNumber = "23239029202"}
+            }.AsQueryable();
+
+            MockSet = new Mock<DbSet<Contact>>();
+            MockSet.As<IQueryable<Contact>>().Setup(m => m.Provider).Returns(Data.Provider);
+            MockSet.As<IQueryable<Contact>>().Setup(m => m.Expression).Returns(Data.Expression);
+            MockSet.As<IQueryable<Contact>>().Setup(m => m.ElementType).Returns(Data.ElementType);
+            MockSet.As<IQueryable<Contact>>().Setup(m => m.GetEnumerator()).Returns(() => Data.GetEnumerator());
+        }
+        [TestMethod]
+        public async Task GetContactReturnsContactsFromDatabase()
```

Comment:
Maybe you could add test for the `public static async Task<IResult> GetContact(IResultImplementationContext context, int id)` to demonstrate both `Ok` and `NotFound`.

Reply:
Ok chief

Reply:
Done